### PR TITLE
Add link to docs artifact to pull requests

### LINKED ISSUE #3572: Provide a way for documentation contributors to preview their documents with the theme from `napari.github.io`
## 🧰 Task
After the migration proposed in #3529 is complete, users will be able to contribute documentation by only making PRs to `napari/napari`. However, because the theme and build tooling all lives at `napari.github.io`, they will not be easily able to preview their documents.

We therefore need a way to build a preview using the theme from `napari.github.io` but pointing to `napari` docs sources. 

The best workflow for the contributor (arguably) is having a local preview they can build that refreshes quickly with any new changes, rather than the approach we've taken on the napari hub of previewing in a GitHub Action.

`napari.github.io` already has the capability of local preview using a `yarn` development environment. @codemonkey800 can expand more on how that works. Ideally we could reuse much of this work to preview based on `napari/napari` docs.

@neuromusic and I were thinking of a workflow whereby you could run `make docs` or something similar in your local `napari/napari`, and the minimal content required to build the theme would then be fetched from `napari.github.io` and used to build the theme and provide you with a local preview. @codemonkey800 how feasible is this? What would be the minimum content required locally to run `yarn dev` or a similar command but point it to `napari/napari/docs` instead of the website repo's docs content?

### PR DESCRIPTION
**Pull request by melissawm**
# Description
This PR adds a workflow to create a comment with a link to the built artifacts to all PRs.

Ideally, this should only be run on docs PRs, but this is a first iteration.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# References
Closes #3572

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
> Ideally, this should only be run on docs PRs, but this is a first iteration.

In theory, almost all PRs should be doc PRs, barring some bug fixes or something. 😉

**Comment by alisterburt** [comment]
Love this @melissawm ! Looks like something isn't quite working with the action https://github.com/napari/napari/runs/8194987117?check_suite_focus=true#step:8:26 

Really excited with where this will get us! 😍

**Comment by melissawm** [comment]
Yup @alisterburt - the documentation on that action is less than ideal but I'll try debugging it, I'll keep this as a draft until I figure it out. Thanks!

**Comment by alisterburt** [comment]
Good luck! 😅

**Comment by melissawm** [comment]
Not sure if this would need to be merged before the workflow is triggered... otherwise this should be working. Are there any CI people who might be able to help here?

File: .github/workflows/artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Comment Artifacts
+on:
+  workflow_run:
+    workflows: [Build Docs]
```
Comment (alisterburt) [inline .github/workflows/artifacts.yml:4]: does this need to match the name `docs` rather than `Build Docs`?

**Review by alisterburt — COMMENTED** [review]
This looks good, not sure why it's not running after force-push though so hesitant to merge... have made a small suggestion but really not sure. cc-ing @jaimergp as a CI expert!

File: .github/workflows/artifacts.yml (hunk 1 above)
Comment (jaimergp) [reply .github/workflows/artifacts.yml:4]: According to the [documentation](https://docs.github.com/en/github-ae@latest/actions/using-workflows/events-that-trigger-workflows#workflow_run), it should be the name (`Build docs`), not the workflow filename. 

Also, per the docs:

> Note: This event will only trigger a workflow run if the workflow file is on the default branch.

So this won't run until it makes it to `main`.

Comment (melissawm) [reply .github/workflows/artifacts.yml:4]: Ah got it, thanks!

**Comment by melissawm** [comment]
So, according to @jaimergp 's comment above, this needs to land on `main` for it to actually execute. So I enabled it on my own fork to get a preview. Here it is:

https://github.com/melissawm/napari/pull/15

**Review by alisterburt — APPROVED** [review]
Approving, thanks @melissawm and @jaimergp !

**Pull request merged**